src/lib/src/x86_arrops.cpp

```diff
@@ -0,0 +1,133 @@
+#include "arch_config.h"
+
+#ifdef DISTOPIA_X86_SSE4_1
+
+#include <cstddef>
+#include <cstdint>
+#include <immintrin.h>
+
+#include "arrops.h"
+#include "compiler_hints.h"
+#include "x86_gintrin.h"
+#include "x86_swizzle.h"
+#include "ops.h"
+
+namespace {
+
+template <typename T, typename U>
+constexpr bool IsAligned(const U* addr) {
```

Comment:
What alignment width does this check? `__mm128` requires 16 bit while `__mm256` requires 32 bit? 

This brings up another point is that should we make our own allocator that uses `aligned_alloc ` or something similar? Pain having to do it by hand all the time and there is no way to do it from the cython level (as far as I know).

Reply:
Also brings up the question as to whether we gain much from using `load` vs `loadu` and `store` vs `storeu`. The intrinsics guide lists the same latency, but more a question of `movups` vs `movaps` etc...

Reply:
There is no longer any benefit to using aligned loads/stores. If your pointer is aligned, then `load` and `loadu` (and `store` and `storeu`) are equally fast.

The reason why there are two versions is that early processors with SSE had a penalty for using unaligned instructions on aligned data. This is no longer the case so we can always use the unaligned version.

That said, working with aligned data is still faster (just no need to specifically use the aligned instructions) and the `stream` instruction for streaming stores requires an aligned pointer.

Reply:
All I was thinking is just that we internally enforce the use of aligned_alloc rather than malloc or new, but I'm not sure. Perhaps @richardjgowers will have an opinion?

Reply:
I sure am in favour of using `aligned_alloc` internally. User-provided buffers are a different question.

Reply:
makes sense.